### PR TITLE
feat: twitter nft project present

### PR DESCRIPTION
**Pull request by hanszhoou**
## Description
https://mask.atlassian.net/browse/MF-2873
https://mask.atlassian.net/browse/MF-2211
https://mask.atlassian.net/browse/MF-2893

Closes # (NO_ISSUE)

## Type of change

- [ ] Documentation
- [ ] Code refactoring (Restructuring existing code w/o changing its observable behavior)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (a fix or feature that would make something no longer possible to do/require old user must upgrade their Mask Network to this new version)

## Checklist

- [ ] My code follows the style guidelines of this project.
- [ ] I have performed a self-review of my own code.
  - [ ] I have removed all in development `console.log`s
  - [ ] I have removed all commented code.
- [ ] I have commented on my code, particularly in hard-to-understand areas.
- [ ] I have read [Internationalization Guide](https://github.com/DimensionDev/Maskbook/blob/develop/docs/i18n-guide.md) and moved text fields to the i18n JSON file.

If this PR depends on external APIs:

- [ ] I have configured those APIs with CORS headers to let extension requests get passed. 
  - chrome extension: `chrome-extension://[id]`
  - firefox extension: `moz-extension://[id]`
- [ ] I have delegated all web requests to the background service via the internal RPC bridge.

### CODE REVIEW

File: packages/mask/src/plugins/Trader/SNSAdaptor/trending/TrendingPopper.tsx [hunk 1]
```diff
@@ -43,11 +50,13 @@ export function TrendingPopper(props: TrendingPopperProps) {
...
     useEffect(
         () =>
-            PluginTraderMessages.cashTagObserved.on((ev) => {
+            PluginTraderMessages.cashAnchorObserved.on((ev) => {
                 const update = () => {
                     setLocked(true)
                     setName(ev.name)
                     setType(ev.type)
+                    setAddress(ev.address ?? '')
+                    setIsNFTProjectPopper(Boolean(ev.isNFTProjectPopper))
```
Comment (guanbinrui) [inline packages/mask/src/plugins/Trader/SNSAdaptor/trending/TrendingPopper.tsx:59]:
```suggestion
                    setIsNFTProjectPopper(ev.isNFTProjectPopper)
```

File: packages/mask/src/plugins/Trader/messages.ts [hunk 2]
```diff
@@ -1,12 +1,16 @@
...
 
-interface CashTagEvent {
+interface CashAnchorEvent {
     name: string
     type: TagType
-    element: HTMLAnchorElement | null
+    element: HTMLElement | null
+    dataProviders?: SourceType[]
+    address?: string
+    isNFTProjectPopper?: boolean
```
Comment (guanbinrui) [inline packages/mask/src/plugins/Trader/messages.ts:13]: How can a cash anchor event collect that information?

File: packages/mask/src/components/InjectedComponents/ProfileTabContent.tsx [hunk 3]
```diff
@@ -178,6 +180,12 @@ export function ProfileTabContent(props: ProfileTabContentProps) {
         })
     }, [retryIdentity])
 
+    useEffect(() => {
+        return MaskMessages.events.NFTProjectTwitterDetect.on((ev) => {
```
Comment (guanbinrui) [inline packages/mask/src/components/InjectedComponents/ProfileTabContent.tsx]: Please don't make it depend on a remote event. Can we receive sth as a prop and derive it into an address?

File: packages/mask/src/plugins/Trader/SNSAdaptor/NFTProjectAvatarBadge.tsx [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+
+    useEffect(() => {
+        const button = buttonRef.current
+        if (!button) return
+
+        let openTimer: NodeJS.Timeout
+        const enter = () => {
+            clearTimeout(openTimer)
+
+            const button = buttonRef.current
```
Comment (guanbinrui) [inline packages/mask/src/plugins/Trader/SNSAdaptor/NFTProjectAvatarBadge.tsx]: Avoid shadowing an outside variable.

File: packages/mask/src/plugins/Trader/trending/useTrending.ts [hunk 5]
```diff
@@ -12,6 +22,54 @@ import { useCurrentCurrency } from './useCurrentCurrency.js'
...
+    }, [twitterHandler])
+}
+
+export function useNonFungibleTokenActivities(address: string, expectedChainId?: Web3Helper.ChainIdAll) {
+    const pageIndexRef = useRef<number>(0)
+    const [nonFungibleTokenActivities, setNonFungibleTokenActivities] = useState<
+        Record<number, NonFungibleTokenActivity[]>
+    >({})
+
+    const getNonFungibleTokenActivities = useCallback(async () => {
```
Comment (guanbinrui) [inline packages/mask/src/plugins/Trader/trending/useTrending.ts]: Refactor to use `useAsyncRetry`.

File: packages/web3-providers/src/NFTScan/helpers/EVM.ts [hunk 6]
```diff
@@ -25,14 +25,15 @@ import {
...
+import { NFTSCAN_BASE, NFTSCAN_LOGO_BASE, NFTSCAN_URL, NFTSCAN_API, NFTSCAN_RESTFUL_API } from '../constants.js'
 import type { EVM } from '../types/EVM.js'
 import { getAssetFullName } from '../../helpers/getAssetFullName.js'
 import { resolveActivityType } from '../../helpers/resolveActivityType.js'
 import { getPaymentToken } from '../../helpers/getPaymentToken.js'
 import { parseJSON } from '../../helpers/parseJSON.js'
+import type { NonFungibleTokenAPI } from '../../entry-types.js'
 
-type NFTScanChainId =
+export type NFTScanChainId =
```
Comment (guanbinrui) [inline packages/web3-providers/src/NFTScan/helpers/EVM.ts]: Refactor to use `ChainIdEVM`.

File: packages/mask/src/plugins/Trader/messages.ts (hunk 2 above)
Comment (hanszhoou) [reply packages/mask/src/plugins/Trader/messages.ts:13]: renamed cashAnchorObserved => trendingAnchorObserved

File: packages/mask/src/plugins/Trader/SNSAdaptor/trending/TrendingPopper.tsx (hunk 1 above)
Comment (hanszhoou) [reply packages/mask/src/plugins/Trader/SNSAdaptor/trending/TrendingPopper.tsx:59]: isNFTProjectPopper?: boolean

File: packages/mask/src/plugins/Trader/trending/useTrending.ts (hunk 5 above)
Comment (hanszhoou) [reply packages/mask/src/plugins/Trader/trending/useTrending.ts]: Refactored to use `useAsyncFn`

File: packages/mask/src/plugins/Trader/SNSAdaptor/trending/NonFungibleTickersTable.tsx [hunk 7]
```diff
@@ -0,0 +1,245 @@
...
+    useScrollBottomEvent(containerRef, fetchMore)
+    const headCellMap: Record<Cells, string> = {
+        nft: t('plugin_trader_table_nft'),
+        method: t('plugin_trader_table_method'),
+        value: t('plugin_trader_table_value'),
+        from: t('plugin_trader_table_from'),
+        to: t('plugin_trader_table_to'),
+        time: t('plugin_trader_table_time'),
+    }
+    const columns: Cells[] = ['nft', 'method', 'value', 'from', 'to', 'time']
```
Comment (guanbinrui) [inline packages/mask/src/plugins/Trader/SNSAdaptor/trending/NonFungibleTickersTable.tsx]:
```suggestion
    const columns: Array<keyof typeof headCellMap> = ['nft', 'method', 'value', 'from', 'to', 'time']
```

File: packages/web3-providers/src/types/Trending.ts [hunk 8]
```diff
@@ -49,7 +48,7 @@ export namespace TrendingAPI {
     export type CommunityUrls = Array<{ type: Partial<CommunityType>; link: string }>
     export interface Coin {
```
Comment (guanbinrui) [inline packages/web3-providers/src/types/Trending.ts]:
```suggestion
    export interface Coin<ChainId> {
```

File: packages/web3-providers/src/NFTScan/helpers/EVM.ts [hunk 9]
```diff
@@ -71,6 +71,75 @@ export async function fetchFromNFTScanV2<T>(chainId: ChainId, pathname: string,
...
+        },
+        cache: 'no-cache',
+    })
+
+    const json = await response.json()
+    return json as T
+}
+
+function resolveNFTScanAPIChain(chainId: ChainId): string {
+    switch (chainId) {
```
Comment (guanbinrui) [inline packages/web3-providers/src/NFTScan/helpers/EVM.ts:107]: Read the `shortName` of `NETWORK_DESCRIPTORS`.

File: packages/web3-providers/src/NFTScan/apis/TrendingAPI.ts [hunk 10]
```diff
@@ -59,6 +74,40 @@ export class NFTScanTrendingAPI implements TrendingAPI.Provider<ChainId> {
         return response?.data ?? EMPTY_LIST
     }
 
+    async getCollectionOverview(
+        chainId: Web3Helper.ChainIdAll,
```
Comment (guanbinrui) [inline packages/web3-providers/src/NFTScan/apis/TrendingAPI.ts]: You can follow other methods in this class.

```suggestion
        chainId: ChainId,
```

File: packages/web3-providers/src/NFTScan/apis/DSearchAPI.ts [hunk 11]
```diff
@@ -39,3 +39,24 @@ export class NFTScanSearchAPI<ChainId, SchemaType> implements DSearchBaseAPI.Dat
...
+            fetchCached,
+        )
+        const collections = await fetchJSON<Array<NonFungibleCollectionResult<ChainId, SchemaType>>>(
+            collectionsURL,
+            undefined,
+            fetchCached,
+        )
+        return (await Promise.allSettled([collectionsFromSpecialList, collections]))
+            .map((v) => (v.status === 'fulfilled' && v.value ? v.value : []))
+            .flat()
```
Comment (guanbinrui) [inline packages/web3-providers/src/NFTScan/apis/DSearchAPI.ts]:
```suggestion
            .flatMap((v) => (v.status === 'fulfilled' && v.value ? v.value : []))
```

File: packages/mask/src/plugins/Trader/messages.ts (hunk 2 above)
Comment (guanbinrui) [reply packages/mask/src/plugins/Trader/messages.ts:13]: <s>Not the naming problem. It's that we cannot grab that information from the cash links.</s>

We can use a `source` field to present the event source.

File: packages/mask/src/components/InjectedComponents/ProfileTabContent.tsx [hunk 12]
```diff
@@ -286,8 +288,17 @@ export function ProfileTabContent(props: ProfileTabContentProps) {
         })
     }
 
+    const { value: collectionList } = useCollectionByTwitterHandler(currentVisitingUserId)
+    const NFTProjectAddress = collectionList?.[0].address
     if (hidden) return null
 
+    if (NFTProjectAddress)
```
Comment (guanbinrui) [inline packages/mask/src/components/InjectedComponents/ProfileTabContent.tsx]:
```suggestion
    if (hidden) return null

    const NFTProjectAddress = collectionList?.[0].address
    if (NFTProjectAddress)
```

File: packages/mask/src/plugins/ProfileCard/SNSAdaptor/ProfileAvatarBadge.tsx [hunk 13]
```diff
@@ -26,7 +26,7 @@ export const ProfileAvatarBadge: FC<Props> = ({ userId, className, ...rest }) =>
         const enter = () => {
             clearTimeout(openTimer)
             clearTimeout(closeTimer)
-            const button = buttonRef.current
+
             if (!button) return
```
Comment (guanbinrui) [inline packages/mask/src/plugins/ProfileCard/SNSAdaptor/ProfileAvatarBadge.tsx]:
```suggestion
```

File: packages/mask/src/plugins/Trader/SNSAdaptor/NFTProjectAvatarBadge.tsx [hunk 14]
```diff
@@ -0,0 +1,58 @@
...
+
+    useEffect(() => {
+        const button = buttonRef.current
+        if (!button) return
+
+        let openTimer: NodeJS.Timeout
+        const enter = () => {
+            clearTimeout(openTimer)
+
+            if (!button) return
```
Comment (guanbinrui) [inline packages/mask/src/plugins/Trader/SNSAdaptor/NFTProjectAvatarBadge.tsx]:
```suggestion
```

File: packages/web3-providers/src/NFTScan/helpers/EVM.ts (hunk 9 above)
Comment (hanszhoou) [reply packages/web3-providers/src/NFTScan/helpers/EVM.ts:107]: Not match.

**Review by guanbinrui — APPROVED** [review]

**Pull request merged**